relay-redis/src/config.rs

```diff
@@ -1,5 +1,30 @@
 use serde::{Deserialize, Serialize};
 
+const fn default_max_connections() -> u32 {
+    24
+}
+
+/// Additional configuration options for a redis client
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct RedisConfigOptions {
+    /// Maximum number of connections managed by the pool
+    #[serde(default = "default_max_connections")]
+    pub max_connections: u32,
+
+    /// If true, the health of a connection will be verified before it's checked out of the pool
+    #[serde(skip, default)]
+    pub test_on_check_out: bool,
```

Comment:
This behavior was initially introduced as a workaround for suboptimal behavior of the `r2d2` bindings in the `redis` crate. See https://github.com/getsentry/relay/pull/1394

As discussed offline, the proper fix for this would be debouncing in the `redis` crate directly, so we better not make this configurable.